### PR TITLE
chore: Extract `babel-plugin-optimize-i18n` to a separate package

### PR DESCRIPTION
**Pull request by scinos**
#### Changes proposed in this Pull Request

Extracts babel-plugin-optimize-i18n ([original PR](https://github.com/Automattic/wp-calypso/pull/49341)) to a separate package `@automattic/babel-plugin-optimize-i18n`.

The main reason is I'm trying to split `@automattic/calypso-build` into smaller, granular parts that don't require users to pull in all peer dependencies.

#### Testing instructions

Run `rm -fr .cache build public && NODE_ENV=production CALYSPO_ENV=production BUILD_TRANSLATION_CHUNKS=true yarn build-client` in this branch and `trunk` and verify the output is the same (`diff -qr wp-calypso/public/evergreen wp-calypso2/public/evergreen`)

### CODE REVIEW

File: packages/babel-plugin-optimize-i18n/README.md [hunk 1]
```diff
@@ -0,0 +1,25 @@
+# Babel plugin optimize i18n
```
Comment (scinos) [inline packages/babel-plugin-optimize-i18n/README.md:1]: Copied from the [original PR](https://github.com/Automattic/wp-calypso/pull/49341) description

**Comment by noahtallen** [comment]
Why couldn't this just be in @automattic/calypso-babel-config?

**Comment by scinos** [comment]
> Why couldn't this just be in `@automattic/calypso-babel-config`?

It could, yes. My reasons for using a separate package:

- Consistency: Other Babel plugins are already in their own packages: `@automattic/babel-plugin-i18n-calypso`, `@automattic/babel-plugin-transform-wpcalypso-async`
- Cohesion: `@automattic/calypso-babel-config` is about config, not about plugins. I'd like to avoid mixing concerns.

Fundamentally, I think packages are the essential building block of NPM projects, the default composition unit, and are virtually free. Unless there is a very strong, concrete reason not to, we should default to extract packages.

What benefits do you see in having this plugin in `@automattic/calypso-babel-config`?

**Comment by noahtallen** [comment]
> What benefits do you see in having this plugin in @automattic/calypso-babel-config?

I guess it's just subjective, but I think there are downsides for a package consumer:
- Keeping them all up to date at the same time
- Having to add multiple packages to solve a single problem
- It almost feels like exposing the implementation details of the babel config package
- There is some level of cost to maintaining lots of package.json. For example, if we need to adjust the format of any package config file, we have lots of files to work through. (Honestly not that big a deal though.) 

I guess to me, the purpose of adding the package would be so that others could use it, but it seems like this is just a single file that would only ever be used by the babel config. And then beyond that, I see packages as a collection of similar tools -- like having a single components package, rather than a different package for each component.

From my subjective point of view, I like the idea of installing `@automattic/calypso-babel-config`, and then that provides everything I could need for using babel in my project.

Maybe this is a more special case, given that babel plugins are typically more standalone :)

**Comment by scinos** [comment]
>  I like the idea of installing `@automattic/calypso-babel-config`, and then that provides everything I could need for using babel in my project.

That still works, `@automattic/babel-plugin-optimize-i18n` is a dependency of `@automattic/calypso-babel-config`

But now other projects can just depend on `@automattic/babel-plugin-optimize-i18n` if they use `@wordpress/i18n`, and create their own babel config (for example, if they don't want React, Typescript or our opinionated [preset](https://github.com/Automattic/wp-calypso/blob/2055f8aa437086540aee5a0eb0557cdf1586eb42/packages/calypso-babel-config/presets/default.js))

**Review by noahtallen — APPROVED** [review]
Okay :) I'm good moving forward with this approach, especially for the babel plugin. I'm definitely interested to see where this approach will take us down the road with a more modular calypso build tool.

**Comment by anomiex** [comment]
Thanks for doing this, looking forward to it showing up on npmjs.com! I'm working on improving Jetpack's webpack configs, which includes using packages (like this one) directly instead of having to work around choices in calypso-build that don't work for us.

**Comment by jsnajdr** [comment]
Thank you for moving this to a separate package. It was a quick experimental patch for an urgent issue at the time.

About naming: this plugin doesn't "optimize" the `__` calls, but rather the opposite: it deoptimizes them by preventing mangling. Maybe `babel-plugin-preserve-i18n` or `babel-plugin-demangle-i18n` would be a better name, if it's not too late.

**Comment by scinos** [comment]
Renamed in https://github.com/Automattic/wp-calypso/pull/57528

**Pull request merged**